### DISCUSSION TITLE
plugin became REALLY buggy

### DISCUSSION THREAD

**Discussion by quenbyako** [golang/vscode-go#1221] [Help]
So, i used old (until january 2021 update) vscode-go plugin, and it works really well! All stuff was supported including my custom linters, debugger pluggin etc.

Since last major update (it was major update, isn't?), literally **everything** became too buggy and with lags, even fresh vscode instance with vscode-go plugin only.

Here is a few problems i stuck with:
* sometimes linter just stuck: you're updating file, and expect linter warnings updates. but now sometimes plugin just stuck and doesn't update in ANY event: code updating, force gofmt from terminal, custom linters update, restart language server. nothing
* autoimport doesn't work!!! Holy crap, i love this feature! you can save file, and plugin automatically delete unused imports! Also, when i'm typing some package name, i can just automatically add it via choosing hint which i need.
* random errors like `import package not found`... what??? i have in file import like `import "fmt"` and in mainfunc i have `fmt.Println()`, and even go compiler doesn't have issues, but plugin says that a) `fmt` wasn't imported (lol??) and b) `fmt` symbol not found (what????)
*  random gomod troubles: like module not found, wrong module path etc. i'm really nervous about this one
* delve debugger just don't want to start with new version of plugin, i don't know why.
* etc. etc. etc. a lot of problems

So, anybody stuck with these problems too? if yes, how did you fixed it?

**Comment by hyangah** [comment]
What's the version of `gopls` and `dlv`? Most of features you described are provided by those tools, so please make sure to keep them up to date.

Did you see anything suspicious from the logs and notifications when you follow https://github.com/golang/vscode-go/blob/master/docs/troubleshooting.md#troubleshooting?

The only major changes relevant to what you see in [v0.21.0](https://github.com/golang/vscode-go/releases/tag/v0.21.0) and [v0.22.0](https://github.com/golang/vscode-go/releases/tag/v0.22.0) are 
  - gopls is enabled by default in v0.22.0. (otherwise v0.21.1 and v0.22.X should be almost identical) 
  - dedups the lint errors & gopls errors from v0.21.0.
  - there was not much change in debugging features (😭  sadly).

**Reply by quenbyako** [reply]
lmfao, got too strange version number:
```sh
$ gopls version 
version master, built in $GOPATH mode
```
``` sh
$ dlv version
Delve Debugger
Version: 1.5.1
Build: $Id: bca418ea7ae2a4dcda985e623625da727d4525b5 $
```

looks like delve not too fresh (found 1.6 version in git repo), i'll install update, and give you feedback

but gopls... If i understood correctly, go plugin independently build the langserver for its tasks, so i don't know how to correctly install "custom" (read as "official") gopls build: plugin warns that it need to rebuild

**Reply by stamblerre** [reply]
Looks like you are using a very outdated version of `gopls`. Please open the Command Palette (Ctrl+Shift+P) and select "Go: Install/Update Tools" and select `gopls`. That will give you the latest version.

**Reply by quenbyako** [reply]
@stamblerre yeah, i've updated all tools, hope that it will help me.

Will come back after 1-2 days, thanks a lot!

**Reply by quenbyako** [reply]
@stamblerre yeeeeah, works way more better, thanks a lot! Still not perfect, but i really appreciate your help, thanks!

**Reply by quenbyako** [reply]
@stamblerre cancel that, still got deadlocks when i have long code sessions:

``` sh
$ gopls vetrsion
golang.org/x/tools/gopls v0.6.5
    golang.org/x/tools/gopls@v0.6.5 h1:kLt9rD/dWtVdvc8LmdcxagDFih6zxYXREpKSYYZu5KE=
$ dlv version
Delve Debugger
Version: 1.6.0
Build: $Id: 8cc9751909843dd55a46e8ea2a561544f70db34d $
```

more info: sometimes, i see that problems (errors + warnings from gopls and golangci) just stuck with really old file (like 15-30 minutes, or even this file deleted, warnings stil exist). Mostly it happens when i leave vscode for some time or have really long coding sessions like 2-3 hours. Also i need to note you that i have autosave files trigger every 5 seconds, maybe plugin deadlocking when it take too much file updates? idk.

**Comment by quenbyako** [comment]
Another strange error: since update, sometimes compiler warns me that packages `"fmt"`, `"errors"` or `"sync"` and other stdlib packages not found (not kidding) even if compiler build code successfully

**Comment by stamblerre** [comment]
@quenbyako: Can you please file a new issue (https://github.com/golang/vscode-go/issues/new/choose) following the steps in the [Troubleshooting guide](https://github.com/golang/vscode-go/blob/master/docs/troubleshooting.md)? I don't think GitHub discussions are necessarily the right place for us to investigate the problem. Please make sure you include a `gopls` log in the issue.

**Reply by quenbyako** [reply]
@stamblerre i started discussion cause i don't want to warn contributors if i did something wrong)

Sure, i'll write issue with full descriptions and logs, thanks for feedback!
